src/search.cpp

```diff
@@ -1091,11 +1091,16 @@ namespace {
       if (   !SpNode
           &&  Threads.size() >= 2
           &&  depth >= Threads.minimumSplitDepth
+          &&  thisThread->splitPointsSize < MAX_SPLITPOINTS_PER_THREAD
           &&  (   !thisThread->activeSplitPoint
                || !thisThread->activeSplitPoint->allSlavesSearching
-               || (   Threads.size() > MAX_SLAVES_PER_SPLITPOINT
-                   && thisThread->activeSplitPoint->slavesMask.count() == MAX_SLAVES_PER_SPLITPOINT))
-          &&  thisThread->splitPointsSize < MAX_SPLITPOINTS_PER_THREAD)
+
+                  // Try to split again if previous split was limited due to
+                  // hit max_slaves_per_splitpoint.
+               || [&]() {
```

Comment:
I'd be a bit more comfortable factoring this out into a function.  Using an inline lambda here feels like packing too much into an if statement.